### DISCUSSION TITLE
Hugo site have no style after deploying on GitHub pages

### DISCUSSION THREAD

**Discussion by h-enk** [thuliteio/doks#1207] [Q&A]
Prceeding from https://discourse.gohugo.io/t/hugo-site-have-no-style-after-deploying-on-github-pages/48900

**Comment by h-enk** [comment]
Please try with this recommended workflow — see [Deploy your Hyas Site to GitHub Pages](https://docs.gethyas.com/guides/deploy/github/)

**Reply by h-enk** [reply]
I noticed you're using a globally (not project locally) installed Hugo — any particular reason?

**Reply by h-enk** [reply]
Make sure to push `hugo_stats.json` to your repo

**Reply by amirfarzamnia** [reply]
No, I was attempting to test various versions of Hugo that are included within our server path, rather than just within a folder.

**Comment by jmooring** [comment]
I was able to publish the site to GH pages after making the following changes:

1\) Change the .gitignore file to:

```text
.hugo_build.lock
node_modules/
public/
resources/
```

2\) Run `npm install` to produce a clean `package-lock.json` file which should _always_ be checked into source control.

3\) Run `hugo` to produce a clean `hugo_stats.json` file which, in almost all setups, should be checked into source control.

4\) Use the [workflow recommended by the GitHub Pages team](https://gohugo.io/hosting-and-deployment/hosting-on-github/#procedure), though I think you should follow the theme's recommendations.

Repository: <https://github.com/jmooring/botstudio-hub>
Live site: <https://jmooring.github.io/botstudio-hub/>

**Comment by jmooring** [comment]
@h-enk Even though <https://jmooring.github.io/botstudio-hub/> looks right, the blog and docs links in the header 404.

**Reply by h-enk** [reply]
Yes, I noticed — thanks, I'll look into it

**Comment by jmooring** [comment]
![image](https://github.com/gethyas/doks/assets/335669/8d3ecac9-868e-4d77-bf54-a496a2fbbbb0)

**Comment by jmooring** [comment]
Disregard. User error.

**Comment by jmooring** [comment]
After setting the baseURL to `https://jmooring.github.io/botstudio-hub/` I'm still seeing:

![image](https://github.com/gethyas/doks/assets/335669/37742b09-0512-4f08-9e38-f22d3b05d702)

**Reply by h-enk** [reply]
Not sure yet what's the issue — this is a working setup: https://github.com/gethyas/doks-gh-pages

**Reply by jmooring** [reply]
OP had canonifyURLs = true in production site config. Changed to false (the default) and everything seems to be working fine. That config setting is likely to be deprecated in a future version. There is never a good reason to use it.

**Reply by h-enk** [reply]
Ah, great that's good to know!

**Reply by amirfarzamnia** [reply]
It seems like the issue has been resolved. That's fantastic news!

**Reply by h-enk** [reply]
Nice site! https://hub.botstudioo.com/ Super thanks @jmooring

**Reply by amirfarzamnia** [reply]
Thanks to you!

**Reply by pranavdhawale** [reply]
Hey guys,

I read your discussions but still i am facing this issue. I removed canonifyUrl field but still the website shows not styling at all.
Help will be appreciated.

![image](https://github.com/gethyas/doks/assets/74247818/68f8ddda-cc90-4e45-9b94-f314d8b8d478)

**Reply by amirfarzamnia** [reply]
Here is an example you can use: https://github.com/absyro/botstudio-hub

**Reply by h-enk** [reply]
Here's another (updated) example repo: https://github.com/gethyas/doks-gh-pages

The (updated) [Deploy your Hyas Site to GitHub Pages](https://docs.gethyas.com/guides/deploy/github/) guide has step-by-step instructions for deploying your Doks/Hyas site to GitHub pages.

👉 Do NOT set `canonifyURLs = true` — see also https://github.com/gethyas/doks/discussions/1240

**Reply by pranavdhawale** [reply]
Thanks guys!! Was able to successfully deploy the website.

**Reply by pranavdhawale** [reply]
Also guys, whats with the Categories and Tags page?? The text isn't visible only there are multiple block with arrows.
![image](https://github.com/gethyas/doks/assets/74247818/4adbbe7f-dd08-48f2-ba21-19a97cf0fde4)

And can i make my own endpoints like different pages other than docs, blogs, categories, tags.
How can i disable any already present endpoint, Eg. I want to disable the categories page

**Reply by h-enk** [reply]
> And can i make my own endpoints like different pages other than docs, blogs, categories, tags.

Just create content (and set `draft: false` in the frontmatter) — that's it

See also the Hugo docs:  [disableKinds](https://gohugo.io/getting-started/configuration/#disablekinds) and [Build options](https://gohugo.io/content-management/build-options/)

**Reply by pranavdhawale** [reply]
>Also guys, whats with the Categories and Tags page?? The text isn't visible only there are multiple block with arrows.

> How can i disable any already present endpoint, Eg. I want to disable the categories page

Please help me out

**Reply by h-enk** [reply]
See [How to remove the pages that list all categories and tags](https://discourse.gohugo.io/t/how-to-remove-the-pages-that-list-all-categories-and-tags/29337)

**Reply by pranavdhawale** [reply]
Now my search is not working 😭 😭. And i have no idea why. It was working earlier before working on cover image. 

Here is my repo link could you please checkout the issue
Repo: [ByteBlogs/byteblogs](https://github.com/ByteBlogs/byteblogs)
Live: [ByteBlogs](https://github.com/ByteBlogs/byteblogs)

**Reply by h-enk** [reply]
You found a bug!

For now, copy `node_modules/@hyas/doks-core/assets/js/flexsearch.js` to `assets/js/flexsearch.js` and comment out line 126 and add below `fetch("{{ relLangURL "search-index.json" }}")`:

```js
  function buildIndex() {
    document.querySelector('.search-loading').classList.remove('d-none');
    // fetch("{{ site.LanguagePrefix }}/search-index.json")
    fetch("{{ relLangURL "search-index.json" }}")
      .then(function (response) {
        return response.json();
      })
      .then(function (data) {
        data.forEach(function (item) {
          index.add(item);
        });
      });
  }
```

**Reply by h-enk** [reply]
https://github.com/gethyas/doks-core/issues/91

**Comment by h-enk** [comment]
![2024-05-22_20-47-52](https://github.com/gethyas/doks/assets/3902872/af1de652-2381-46a0-a8ba-61ddc715ba5d)

What's not working?

**Reply by pranavdhawale** [reply]
> For now, copy node_modules/@hyas/doks-core/assets/js/flexsearch.js to assets/js/flexsearch.js and comment out line 126 and add below fetch("{{ relLangURL "search-index.json" }}"):

did this then it started working.... before that it showed no results found

**Reply by h-enk** [reply]
Ah, ok — so the fix works

**Reply by pranavdhawale** [reply]
yes
